service/internal/workflow/workflow.go

```diff
@@ -47,30 +47,39 @@ type Logs struct {
 }
 
 // List returns a list of workflows.
-func (a ArgoWorkflow) List(ctx context.Context) ([]string, error) {
-	workflowIDs := []string{}
+func (a ArgoWorkflow) List(ctx context.Context) ([]Status, error) {
+	workflows := []Status{}
 
 	workflowListResult, err := a.svc.ListWorkflows(ctx, &argoWorkflowAPIClient.WorkflowListRequest{
 		Namespace: a.namespace,
 	})
-
 	if err != nil {
-		return workflowIDs, err
+		return workflows, err
 	}
 
-	for _, item := range workflowListResult.Items {
-		workflowIDs = append(workflowIDs, item.ObjectMeta.Name)
+	for _, wf := range workflowListResult.Items {
```

Comment:
Out of curiosity, can this be a fairly large list? Since this is an optimizing PR, if that list can be large, it might be worth declaring the workflows var after items are returned, making it a preallocated list size, and then direct assignment by index and not using append down on line 71.

Reply:
It is worth declaring :smile: Thanks for catching that.